geode-core/src/distributedTest/java/org/apache/geode/management/internal/pulse/TestFunctionsDUnitTest.java

```diff
@@ -17,26 +17,52 @@
 import static org.junit.Assert.assertNotNull;
 import static org.junit.Assert.assertTrue;
 
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import org.awaitility.Awaitility;
 import org.junit.Test;
 
 import org.apache.geode.cache.Cache;
 import org.apache.geode.cache.execute.Execution;
 import org.apache.geode.cache.execute.Function;
+import org.apache.geode.cache.execute.FunctionContext;
 import org.apache.geode.cache.execute.FunctionService;
-import org.apache.geode.internal.cache.functions.TestFunction;
 import org.apache.geode.management.DistributedSystemMXBean;
 import org.apache.geode.management.ManagementService;
 import org.apache.geode.management.ManagementTestBase;
-import org.apache.geode.test.dunit.LogWriterUtils;
+import org.apache.geode.test.dunit.Invoke;
 import org.apache.geode.test.dunit.SerializableRunnable;
 import org.apache.geode.test.dunit.VM;
-import org.apache.geode.test.dunit.Wait;
-import org.apache.geode.test.dunit.WaitCriterion;
 
 /**
  * This is for testing running functions
  */
 
+
+class SimpleWaitFunction<T> implements Function<T> {
+  private static CountDownLatch countDownLatch;
+
+  public void setLatch() {
+    countDownLatch = new CountDownLatch(1);
+  }
+
+  @Override
+  public void execute(FunctionContext<T> context) {
+    T args = context.getArguments();
+    if (args == Boolean.TRUE) {
+      try {
+        countDownLatch.await();
+      } catch (InterruptedException e) {
+        e.printStackTrace();
```

Comment:
Don't catch InterruptedException without rethrowing something or otherwise handling in. In this case, you probably want to fail the test or throw a RuntimeException.